pkg/dataobj/sections/internal/columnar/metrics.go

```diff
@@ -0,0 +1,301 @@
+package columnar
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/grafana/loki/v3/pkg/dataobj"
+	"github.com/grafana/loki/v3/pkg/dataobj/internal/metadata/datasetmd/v2"
+	"github.com/grafana/loki/v3/pkg/dataobj/internal/result"
+)
+
+// Metrics instruments the a columnar section. Metrics are only updated when
+// calling [Metrics.Observe].
+type Metrics struct {
+	columnMetadataSize      prometheus.Histogram
+	columnMetadataTotalSize prometheus.Histogram
+
+	columnCount             prometheus.Histogram
+	columnCompressedBytes   *prometheus.HistogramVec
+	columnUncompressedBytes *prometheus.HistogramVec
+	columnCompressionRatio  *prometheus.HistogramVec
+	columnRows              *prometheus.HistogramVec
+	columnValues            *prometheus.HistogramVec
+
+	pageCount             *prometheus.HistogramVec
+	pageCompressedBytes   *prometheus.HistogramVec
+	pageUncompressedBytes *prometheus.HistogramVec
+	pageCompressionRatio  *prometheus.HistogramVec
+	pageRows              *prometheus.HistogramVec
+	pageValues            *prometheus.HistogramVec
+}
+
+// NewMetrics creates a new set of metrics for the streams section. Call
+// [Metrics.Observe] to update the metrics for a given section.
+func NewMetrics(sectionType dataobj.SectionType) *Metrics {
+	// Create the name without the version specified.
+	sectionName := sectionType.Namespace + "/" + sectionType.Kind
+	sectionLabels := prometheus.Labels{"section": sectionName}
+
+	return &Metrics{
+		columnMetadataSize: newNativeHistogram(prometheus.HistogramOpts{
+			Namespace: "loki_dataobj",
```

Comment:
Minor nit: I believe there was a discussion about Namespace, Subsystem being deprecated and just using name so metrics were more greppable.